imagepruner/v1/types.go

```diff
@@ -0,0 +1,97 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConfigList is a slice of Config objects.
+type ConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Config `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config is the configuration object for an image registry pruner
+// managed by the registry operator.
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImagePrunerSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImagePrunerStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// PrunerHistory is a configuration object for how many
+// successful and failed finished jobs to retain.
+type PrunerHistory struct {
+	// SuccessfulJobsHistoryLimit specifies how many successful finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit" protobuf:"bytes,1,opt,name=successfulJobsHistoryLimit"`
+	// FailedJobsHistoryLimit specifies how many failed finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit" protobuf:"bytes,2,opt,name=failedJobsHistoryLimit"`
+}
+
+type ImagePrunerSpec struct {
+	// Schedule specifies when to execute the job
+	// Uses standard cronjob syntax: https://wikipedia.org/wiki/Cron
+	// Defaults to 0 0 * * *
+	// +optional
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+	// Suspend specifies wether or not to suspend subsequent executions of this cronjob.
+	// Defaults to false
+	// +required
+	Suspend *bool `json:"suspend" protobuf:"bytes,2,opt,name=suspend"`
+	// KeepTagRevisions specifies how many tag revisions to keep.
+	// Defaults to 5
+	// +optional
+	KeepTagRevisions *int `json:"keepTagRevisions" protobuf:"bytes,3,opt,name=keepTagRevisions"`
+	// KeepYoungerThan specifies how old an image needs to be for it to be pruned.
+	// Defaults to 96h (96 hours)
+	// +optional
+	KeepYoungerThan string `json:"keepYoungerThan" protobuf:"bytes,4,opt,name=keepYoungerThan"`
+	// Resources defines the resource requests+limits for the registry pod.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources" protobuf:"bytes,5,opt,name=resources"`
```

Comment:
I believe if you are using pointers, optional fields should have `omitempty`:
```suggestion
	Resources *corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,5,opt,name=resources"`
```

Reply:
updated in next push